src/VM/Handlers/Opcode8046Handler.cpp

```diff
@@ -35,9 +37,21 @@ Opcode8046Handler::Opcode8046Handler(VM* vm) : OpcodeHandler(vm)
 
 void Opcode8046Handler::_run()
 {
-    Logger::debug("SCRIPT") << "[8046] [*] - value (change sign)" << std::endl;
-    auto value = _vm->popDataInteger();
-    _vm-> pushDataInteger(-value);
+    Logger::debug("SCRIPT") << "[8046] [*] op_negate" << std::endl;
+    auto value = _vm->dataStack()->pop();
+    if (value->type() == VMStackValue::TYPE_INTEGER) 
+    {
+        _vm->pushDataInteger(- (dynamic_cast<VMStackIntValue*>(value))->value());
+    }
+    else
+    if (value->type() == VMStackValue::TYPE_FLOAT) 
+    {
+        _vm->pushDataFloat(- (dynamic_cast<VMStackFloatValue*>(value))->value());
+    }
+    else
+    {
```

Comment:
Then possibly have a new level in logger, say `Logger::script`. It may need the ability to be toggled on/off regardless of other states.